editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -249,14 +247,37 @@ export const previewStore: StateCreator<
           {} as Store.passport["data"],
         );
 
-        return {
+        const passport: Store.passport = {
           ...acc,
           data: {
             ...acc.data,
             ...responseData,
             ...passportData,
           },
         };
+
+        const isSetValue = flow[id].type === TYPES.SetValue;
+
+        if (isSetValue) {
+          const { operation, fn } = flow[id]?.data as SetValue;
+          const previousValues = acc.data?.[fn] || [];
+          const currentValue = responseData?.[fn] || [];
+
+          if (operation === "remove") {
+            const removeCurrentValue = (val: string) => val !== currentValue[0];
+            const filtered = previousValues.filter(removeCurrentValue);
```

Comment:
_Edge Case_:

I don't think it's very likely to happen but if a user wants to `append` or `remove` a passport `fn` which is previously not an array there's an error thrown:

<img width="800" alt="Screenshot 2024-03-25 at 15 58 34" src="https://github.com/theopensystemslab/planx-new/assets/36415632/587800cc-cd0f-4c03-ab12-c777786728e6">

<img width="934" alt="Screenshot 2024-03-25 at 16 00 32" src="https://github.com/theopensystemslab/planx-new/assets/36415632/750d8885-ae43-42ca-888d-b32c97e195bc">

Not very likely to happen but I thought I'd flag it.

Reply:
Very good catch! Please see [b598d7e](https://github.com/theopensystemslab/planx-new/pull/2914/commits/b598d7eb821c6777acd41249b4417f942b03ea67)

To follow - tests for non-arrays.

Reply:
This has now been resolved!